akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -636,25 +684,31 @@ private[akka] class RemoteActorRef private[akka] (
   /**
    * Determine if a watch/unwatch message must be handled by the remoteWatcher actor, or sent to this remote ref
    */
-  def isWatchIntercepted(watchee: ActorRef, watcher: ActorRef) = {
+  def isWatchIntercepted(watchee: ActorRef, watcher: ActorRef): Boolean = {
     // If watchee != this then watcher should == this. This is a reverse watch, and it is not intercepted
     // If watchee == this, only the watches from remoteWatcher are sent on the wire, on behalf of other watchers
-    watcher != provider.remoteWatcher && watchee == this
+    provider.remoteWatcher.exists(remoteWatcher => watcher != remoteWatcher) && watchee == this
   }
 
   def sendSystemMessage(message: SystemMessage): Unit =
     try {
       //send to remote, unless watch message is intercepted by the remoteWatcher
       message match {
-        case Watch(watchee, watcher) if isWatchIntercepted(watchee, watcher) =>
```

Comment:
The bug is here, in the original the `if` is in the cases so that the second watch from `remote-watcher` was done by the `case _`. Now we just swallow that.